#This patch file is from Qualcomm Innovation Center, Inc. and is provided under the following license:
#
#Copyright (c) 2025 Qualcomm Innovation Center, Inc. All rights reserved.
#SPDX-License-Identifier: BSD-3-Clause-Clear
From c40c094a453b6c8577724120941b0541dbbc4015 Mon Sep 17 00:00:00 2001
From: weijshen <quic_weijshen@quicinc.com>
Date: Tue, 29 Apr 2025 14:56:46 +0800
Subject: [PATCH] {jazzy}: remove recipes from blacklist

Remove navigation2, nav2-bringup, moveit-planners-ompl, moveit-planners, moveit-ros-perception 
and moveit-runtime from blacklist.

Upstream-Status: Pending

Signed-off-by: weijshen <quic_weijshen@quicinc.com>
---
 .../include/jazzy/ros-distro-recipe-blacklist.inc         | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc b/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc
index d42023e9a..890e3b107 100644
--- a/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc
+++ b/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc
@@ -44,20 +44,16 @@ SKIP_RECIPE[joint-state-publisher-gui] ?= "${@bb.utils.contains_any('ROS_WORLD_S
 SKIP_RECIPE[libg2o] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['opengl', 'x11'], 'opengl: depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: depends on libglu which requires x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[librealsense2] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['opengl', 'glfw'], 'opengl: depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; glfw: depends on librealsense2 which depends on glfw which is available only in dunfell and requires x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[moveit] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'opengl', 'x11', 'libomp'], 'qt5: rdepends on moveit-ros which rdepends on moveit-ros-visualization which depends on rviz2 which depends on qtbase; opengl: rdepends on moveit-ros rdepends on moveit-ros-visualization which depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: rdepends on moveit-ros rdepends on moveit-ros-visualization which depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on libx11,libxrandr,libxaw which require x11 in DISTRO_FEATURES; libomp: rdepends on moveit-planners which rdepends on moveit-planners-ompl which depends on ROS_UNRESOLVED_DEP-libomp-dev which can be provided by meta-clang, but not yet tested with meta-ros', '', d)}"
-SKIP_RECIPE[moveit-planners] ?= "${@bb.utils.contains('ROS_WORLD_SKIP_GROUPS', ['libomp'], 'libomp: rdepends on moveit-planners-ompl which depends on ROS_UNRESOLVED_DEP-libomp-dev which can be provided by meta-clang, but not yet tested with meta-ros', '', d)}"
-SKIP_RECIPE[moveit-planners-ompl] ?= "${@bb.utils.contains('ROS_WORLD_SKIP_GROUPS', ['libomp'], 'libomp: depends on ROS_UNRESOLVED_DEP-libomp-dev which can be provided by meta-clang, but not yet tested with meta-ros', '', d)}"
 SKIP_RECIPE[moveit-resources] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'pyqt5'], 'qt5: rdepends on moveit-resources-panda-moveit-config which rdepends on joint-state-publisher-gui which depends on python-qt-binding which depends on qtbase; pyqt5: rdepends on moveit-resources-panda-moveit-config which rdepends on joint-state-publisher-gui which depends on python-qt-binding which depends on python3-pyqt5 which requires pyqt5', '', d)}"
 SKIP_RECIPE[moveit-resources-panda-moveit-config] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'pyqt5'], 'qt5: rdepends on joint-state-publisher-gui which depends on python-qt-binding which depends on qtbase; pyqt5: rdepends on joint-state-publisher-gui which depends on python-qt-binding which depends on python3-pyqt5 which requires pyqt5', '', d)}"
 SKIP_RECIPE[moveit-ros] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'opengl', 'x11'], 'qt5: rdepends on moveit-ros-visualization which depends on rviz2 which depends on qtbase; opengl: rdepends on moveit-ros-visualization which depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: rdepends on moveit-ros-visualization which depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on libx11,libxrandr,libxaw which require x11 in DISTRO_FEATURES', '', d)}"
-SKIP_RECIPE[moveit-ros-perception] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['opengl', 'x11'], 'Depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES and freeglut which requires x11', '', d)}"
 SKIP_RECIPE[moveit-ros-visualization] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'opengl', 'x11'], 'qt5: depends on rviz2 which depends on qtbase; opengl: depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on libx11,libxrandr,libxaw which require x11 in DISTRO_FEATURES', '', d)}"
-SKIP_RECIPE[moveit-runtime] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'opengl', 'x11', 'libomp'], 'qt5: rdepends on moveit-ros which rdepends on moveit-ros-visualization which depends on rviz2 which depends on qtbase; opengl: rdepends on moveit-ros rdepends on moveit-ros-visualization which depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: rdepends on moveit-ros rdepends on moveit-ros-visualization which depends on rviz2 which depends on rviz-rendering which depends on rviz-ogre-vendor which depends on libx11,libxrandr,libxaw which require x11 in DISTRO_FEATURES; libomp: rdepends on moveit-planners which rdepends on moveit-planners-ompl which depends on ROS_UNRESOLVED_DEP-libomp-dev which can be provided by meta-clang, but not yet tested with meta-ros', '', d)}"
 SKIP_RECIPE[moveit-visual-tools] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['opengl', 'x11'], 'opengl: depends on rviz-visual-tools which depends on rviz-ogre-vendor which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: depends on rviz-visual-tools which depends on rviz-ogre-vendor which depends on libx11,libxrandr,libxaw which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[mrpt2] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['ffmpeg', 'openni', 'x264', 'x11'], 'ffmpeg: Depends on ffmpeg which requires commercial license; openni: Depends on libopenni2-dev which is not provided; x264: Depends on ffmpeg which depends on x264 which requires commercial license; x11: Depends on libxxf86vm, freeglut, libxrandr, wxwidgets, glfw which require x11 in DISTRO_FEATURES', '', d)}"
-SKIP_RECIPE[nav2-bringup] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on slam-toolbox which depends on qtbase; x11: depends on slam-toolbox which depends on rviz-common->rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
+#SKIP_RECIPE[nav2-bringup] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on slam-toolbox which depends on qtbase; x11: depends on slam-toolbox which depends on rviz-common->rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[nav2-rviz-plugins] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on qtbase; x11: depends on rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[nav2-system-tests] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on nav2-bringup which depends on slam-toolbox which depends on qtbase; x11: depends on nav2-bringup which depends on slam-toolbox which depends on rviz-common->rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
-SKIP_RECIPE[navigation2] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: rdepends on nav2-rviz-plugins which depends on qtbase; x11: rdepends on nav2-rviz-plugins which depends on rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
+#SKIP_RECIPE[navigation2] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: rdepends on nav2-rviz-plugins which depends on qtbase; x11: rdepends on nav2-rviz-plugins which depends on rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[octomap-rviz-plugins] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on qtbase; x11: depends on rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[octovis] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'libqglviewer'], 'qt5: depends on qtbase which requires meta-qt5 to be included; libqglviewer: depends on libqglviewer2-qt5 which is not available', '', d)}"
 SKIP_RECIPE[openvslam] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['opengl', 'x11'], 'opengl: depends on libg2o which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: depends on libg2o which depends on libglu which requires x11 in DISTRO_FEATURES', '', d)}"
--
2.34.1

